examples/ssr/src/entry-server.js

```diff
@@ -0,0 +1,41 @@
+import { createApp } from './main';
+
+export default context =>
+  new Promise(async (resolve, reject) => {
+    const { app, router, instantsearch } = await createApp();
+
+    router.push(context.url);
+
+    router.onReady(() => {
+      const matchedComponents = router.getMatchedComponents();
+
+      if (!matchedComponents.length) {
+        return reject({ code: 404 });
+      }
+
+      Promise.all(
+        matchedComponents.map(Component => {
+          if (Component.asyncData) {
+            return Component.asyncData({
+              instantsearch,
+              route: router.currentRoute,
+            });
+          }
+        })
+      )
+        .then(components =>
+          instantsearch.findRoot({ components, context })
```

Comment:
We can avoid this function IMO - we can expose a function like `getState()` whatever the name to retrieve the results that we get from `asyncData`. This is how Vuex works, so we might expect that people know what to do. It would looks like:

```js
Promise.all(
  matchedComponents.map(Component => {
    if (Component.asyncData) {
      return Component.asyncData({
        instantsearch,
        route: router.currentRoute
      });
    }
  })
).then(() => {
  // Probably not the best name but you got the idea
  context.instantsearchState = instantsearch.getState();

  resolve(app);
});
```